File: src/main/java/org/sylfra/idea/plugins/remotesynchronizer/ui/ThreadConsoleToolWindowFactory.java
package org.sylfra.idea.plugins.remotesynchronizer.ui;

import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.Content;
import com.intellij.ui.content.ContentFactory;
import org.jetbrains.annotations.NotNull;
import org.sylfra.idea.plugins.remotesynchronizer.FileSyncPlugin;

import javax.swing.*;

/**
 * <p>
 * date: 2023/2/12 16:45
 *
 * @author Zhengyu Hu
 */
public class ThreadConsoleToolWindowFactory implements ToolWindowFactory, DumbAware {

    @Override
    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        FileSyncPlugin fileSyncPlugin = FileSyncPlugin.getInstance(project);

        ContentFactory contentFactory = ContentFactory.getInstance();
        Content content = contentFactory.createContent(new ToolPanel(fileSyncPlugin.getConsolePane(), fileSyncPlugin.getConfig()),
                FileSyncPlugin.PLUGIN_NAME + " Console", true);
        toolWindow.getContentManager().addContent(content);

        //toolWindow.setIcon(new ImageIcon(FileSyncPlugin.getResource("logo-small.png")));
    }
}

File: src/main/kotlin/org/imyuyu/idea/plugins/filesync/ui/ThreadConsoleToolWindowFactory.kt
package org.imyuyu.idea.plugins.filesync.ui

import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowFactory
import com.intellij.ui.content.ContentFactory
import org.imyuyu.idea.plugins.filesync.adaptedMessage
import org.imyuyu.idea.plugins.filesync.FileSyncPlugin

/**
 *
 *
 * date: 2023/2/12 16:45
 *
 * @author Zhengyu Hu
 */
class ThreadConsoleToolWindowFactory : ToolWindowFactory, DumbAware {


    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        if (project.isDisposed) {
            return
        }

        val fileSyncPlugin = FileSyncPlugin.getInstance(project)
        val contentFactory = ContentFactory.getInstance()
        val content = contentFactory.createContent(
            ToolPanel(fileSyncPlugin.consolePane, fileSyncPlugin.config),
            adaptedMessage("TITLE_CONSOLE"), true
        )
        toolWindow.contentManager.addContent(content)

        //toolWindow.setIcon(new ImageIcon(FileSyncPlugin.getResource("logo-small.png")));
    }

    override fun init(toolWindow: ToolWindow) {
        adaptedMessage("FileSync").let { title ->
            toolWindow.title = title
            toolWindow.stripeTitle = title
        }
    }
}
